ouroboros-consensus/tools/db-convert/Main.hs

```diff
@@ -0,0 +1,275 @@
+              (if requiresNetworkMagic then RequiresMagic else RequiresNoMagic)
+              (toFilePath configFile)
+              genesisHash
+        case econfig of
+          Left err -> throwIO $ MkConfigError err
+          Right config -> validateChainDb dbDir config verbose
+
+convertEpochFile
+  :: EpochSlots
+  -> Path Abs File -- ^ Input
+  -> Path Abs Dir -- ^ Ouput directory
+  -> IO (Either CC.ParseError ())
+convertEpochFile es inFile outDir =
+  let inStream = CC.parseEpochFileWithBoundary es (toFilePath inFile)
+      dbDir = outDir </> [reldir|immutable|]
+      encode = CB.serializeEncoding' . Byron.encodeByronBlock . Byron.ByronBlockOrEBB
+   in do
+        createDirIfMissing True dbDir
+        outFileName <-
+          parseRelFile $ "epoch-" <>
+            drop 2 (toFilePath (filename inFile))
+        outFile <- (dbDir </> outFileName) -<.> "dat"
+        IO.withFile (toFilePath outFile) IO.WriteMode $ \h ->
+          runResourceT $ runExceptT $ S.mapM_ (liftIO . BS.hPut h) . S.map encode $ inStream
+
+validateChainDb
+  :: forall blk. (blk ~ ByronBlockOrEBB ByronConfig)
+  => Path Abs Dir -- ^ DB directory
+  -> CC.Genesis.Config
+  -> Bool -- Verbose
+  -> IO ()
+validateChainDb dbDir cfg verbose =
+  do
+    initialCVS <- runExceptT $ CC.initialChainValidationState cfg
+    case initialCVS of
+      Left err -> throwIO $ InitialChainStateError err
+      Right cvs ->
+        withThreadRegistry $ \registry -> do
+          chaindb <- give protocolMagicId $ give epochSlots $ ChainDB.openDB $ args registry
+          blk <- ChainDB.getTipBlock chaindb
+          putStrLn $ "DB tip: " ++ condense blk
+          ChainDB.closeDB chaindb
+        where
+          initLedgerState = ExtLedgerState
+            { ledgerState = Byron.ByronEBBLedgerState $
+                Byron.ByronLedgerState cvs Seq.empty
+            , ouroborosChainState = Seq.empty
+            }
+          protocolMagicId = CB.unAnnotated . getAProtocolMagicId $ CC.Genesis.configProtocolMagic cfg
+          epochSlots = CC.Genesis.configEpochSlots cfg
+          securityParam = SecurityParam $ CC.unBlockCount k
+          k = CC.Genesis.configK cfg
+          args registry =
+            (ChainDB.defaultArgs @blk (toFilePath dbDir))
+              { ChainDB.cdbGenesis = return initLedgerState
+              , ChainDB.cdbDecodeBlock = Byron.decodeByronBlock epochSlots
+              , ChainDB.cdbDecodeChainState = Byron.decodeByronChainState
+              , ChainDB.cdbDecodeHash = Byron.decodeByronHeaderHash
+              , ChainDB.cdbDecodeLedger = Byron.decodeByronLedgerState
+              , ChainDB.cdbEncodeBlock = Byron.encodeByronBlock
+              , ChainDB.cdbEncodeChainState = Byron.encodeByronChainState
+              , ChainDB.cdbEncodeHash = Byron.encodeByronHeaderHash
+              , ChainDB.cdbEncodeLedger = Byron.encodeByronLedgerState
+              , -- Policy
+              ChainDB.cdbValidation = ImmDB.ValidateAllEpochs
```

Comment:
My OCD prefers:
```haskell
, ChainDB.cdbEncodeLedger = Byron.encodeByronLedgerState
-- Policy
, ChainDB.cdbValidation = ImmDB.ValidateAllEpochs
```

Reply:
As does mine! I've been testing `ormolu`, which I mostly like; this is the one thing I don't!

Reply:
That's the problem I have with formatting tools, there's always at least one thing it does that I don't like :slightly_smiling_face:

Reply:
Hmm, can you fix this? And run `stylish-haskell` for uniformity's sake :slightly_smiling_face: